pkg/util/util_linux.go

```diff
@@ -30,15 +30,25 @@ func Umask(mask int) (int, error) {
 }
 
 func CleanupSocketFile(socketPath string) error {
-	fi, err := os.Stat(socketPath)
-	if err == nil && (fi.Mode()&os.ModeSocket) != 0 {
-		// Remove any socket, stale or not, but fall through for other files
+	socketExists, err := DoesSocketExist(socketPath)
+	if err != nil {
+		return err
+	}
+	if socketExists {
 		if err := os.Remove(socketPath); err != nil {
 			return fmt.Errorf("failed to remove stale socket %s with error: %+v", socketPath, err)
 		}
 	}
+	return nil
+}
+
+func DoesSocketExist(socketPath string) (bool, error) {
+	fi, err := os.Stat(socketPath)
+	if err == nil && (fi.Mode()&os.ModeSocket) != 0 {
```

Comment:
This isn't your change (and maybe we can think about this in a followup), but I wonder if it's worth it to log a warning or return an error if the file is not a socket instead of returning false.  I wouldn't expect creating a listening socket would work if the file already existed and it's not a socket file.

Reply:
Yeah good point, if we dropped something else there with the same name the registrar wouldn't be able to create a socket.